### PR TITLE
Merge dev with G0-5485

### PR DESCRIPTION
**Pull request by prateek-giddh**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
This PR is already approved [here](https://github.com/Walkover-Web-Solution/Giddh-New-Angular4-App/pull/4589) the only change this PR introduces is this [commit](https://github.com/Walkover-Web-Solution/Giddh-New-Angular4-App/commit/7fc6d043f0c8560a8bebf543f5bfb06e4a59f600) and resolves the merge conflict with dev

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

* **Other information**:

### CODE REVIEW

File: apps/web-giddh/src/app/ledger/components/newLedgerEntryPanel/newLedgerEntryPanel.component.ts [hunk 1]
```diff
@@ -259,6 +262,10 @@ export class NewLedgerEntryPanelComponent implements OnInit, OnDestroy, OnChange
             }
         });
 
+        this.store.pipe(select(appState => appState.company), take(1)).subscribe((companyData: CurrentCompanyState) => {
+            this.isTcsTdsApplicable = companyData.isTcsTdsApplicable;
```
Comment (rohit-walkover) [inline apps/web-giddh/src/app/ledger/components/newLedgerEntryPanel/newLedgerEntryPanel.component.ts:266]: Does we need take(1) here  and check nullity case of companyData

File: apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts [hunk 2]
```diff
@@ -486,14 +493,22 @@ export class ProformaInvoiceComponent implements OnInit, OnDestroy, AfterViewIni
                 return;
             }
             // if (stateLoader.show) {
-            //     this.showLoader = true;
```
Comment (rohit-walkover) [inline apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts:489]: remove commented code if not required

File: apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts [hunk 3]
```diff
@@ -2710,6 +2760,17 @@ export class ProformaInvoiceComponent implements OnInit, OnDestroy, AfterViewIni
      * @param invoiceForm
      */
     public submitUpdateForm(invoiceForm: NgForm) {
+        this.generateUpdateButtonClicked.next(invoiceForm);
```
Comment (rohit-walkover) [inline apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts:2763]: Return type and check Doc

File: apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts [hunk 4]
```diff
@@ -4119,7 +4172,10 @@ export class ProformaInvoiceComponent implements OnInit, OnDestroy, AfterViewIni
         // Create a new purchase record
         this.purchaseRecordService.generatePurchaseRecord(request).subscribe((response: BaseResponse<any, PurchaseRecordRequest>) => {
             this.handleGenerateResponse(response, this.invoiceForm);
-        }, () => this._toasty.errorToast('Something went wrong! Try again'));
+        }, () => {
+            this.startLoader(false);
+            this._toasty.errorToast('Something went wrong! Try again')
```
Comment (rohit-walkover) [inline apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts:4177]:
```suggestion
            this._toasty.errorToast('Something went wrong! Try again').    check semicolon and Something went wrong! Try again we can put it in a single variable and use for multiple places 
```

File: apps/web-giddh/src/app/shared/header/header.component.ts [hunk 5]
```diff
@@ -1533,4 +1539,29 @@ export class HeaderComponent implements OnInit, AfterViewInit, OnDestroy, AfterV
...
+     * @memberof HeaderComponent
+     */
+    private checkIfCompanyTcsTdsApplicable(): void {
+        const request: OnboardingFormRequest = {
+            country: (this.activeCompany && this.activeCompany.countryV2) ? this.activeCompany.countryV2.alpha2CountryCode : '',
+            formName: 'otherTaxes'
+        };
+        this.commonService.getOnboardingForm(request).subscribe((response) => {
+            if (response && response.status === 'success' && response.body) {
+                console.log('Response received: ', response);
```
Comment (rohit-walkover) [inline apps/web-giddh/src/app/shared/header/header.component.ts]: remove console

File: apps/web-giddh/src/app/ledger/components/newLedgerEntryPanel/newLedgerEntryPanel.component.html [hunk 6]
```diff
@@ -179,31 +179,28 @@
...
+                            (ngModelChange)="toggleOtherTaxesAsidePane(blankLedger.isOtherTaxesApplicable)" />
+                        <span *ngIf="!blankLedger.isOtherTaxesApplicable">Other Tax ?</span>
+
+                        <div class="clearfix">
+                            <div *ngIf="blankLedger.isOtherTaxesApplicable" class="mrT1 pull-right"
+                                [ngStyle]="{ 'width': totalTdElementWidth + 'px'}">
+                                <label>Other Tax</label>
+                                <span class="d-flex">
+                                    <span class="d-flex align-item-center">
+                                        <a href="javascript: void 0" (click)="toggleOtherTaxesAsidePane(true)">
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/ledger/components/newLedgerEntryPanel/newLedgerEntryPanel.component.html:193]: Please replace javascript: void 0 with javascript:;

**Pull request merged**